src/components/List/List.tsx

```diff
@@ -94,10 +94,6 @@ function List<T extends Item>({
   const [selectAllId] = useState(() => uniqueId('selectall-'));
   const selectAllRef = useRef<HTMLInputElement>(null);
 
-  useDeepCompareEffect(
-    () => onSelect(Array.from(selection.values())),
-    [Array.from(selection.values()), onSelect]
-  );
```

Comment:
Removing this could break code relying on it. Existing usage of List and might expect `onSelect` to be called on first render, when passing in new `selected` items, and passing in a new `onSelect` callback.

Reply:
In that case, code that relies on it needs to be modified.
It does not look right to have `onSelect` called on the first render when no action was done: the parent component passes down the `selected` property so why does it need to know the properties that were just passed down?

Reply:
*the items that were just passed down

Reply:
I agree completely. It could be easy to update the code that relies on this or not. I haven't looked yet.

Reply:
Yes it could potentially break some existing code.